### PR TITLE
Add Knowledge-Enhanced Language Model Pre-training (KELM)

### PR DESCRIPTION
**Pull request by joeddav**
Adds the KELM dataset.

- Webpage/repo: https://github.com/google-research-datasets/KELM-corpus
- Paper: https://arxiv.org/pdf/2010.12688.pdf

### CODE REVIEW

File: datasets/kelm/dataset_infos.json [hunk 1]
```diff
@@ -0,0 +1 @@
...l, "supervised_keys": null, "builder_name": "ag_news", "config_name": "default", "version": {"version_str": "0.0.0", "description": null, "major": 0, "minor": 0, "patch": 0}, "splits": {"train": {"name": "train", "num_bytes": 1343187306, "num_examples": 6371131, "dataset_name": "ag_news"}, "validation": {"name": "validation", "num_bytes": 167790917, "num_examples": 796471, "dataset_name": "ag_news"}, "test": {"name": "test", "num_bytes": 167921750, "num_examples": 796493, "dataset_name": "ag_news"}}, "download_checksums": {"https://storage.googleapis.com/gresearch/kelm-corpus/quadruples-train.tsv": {"num_bytes": 1305075939, "checksum": "55c1d4e1beaccda979fc7193e192bc48af05bf3357bd7c14b93ba750fca91c55"}, "https://storage.googleapis.com/gresearch/kelm-corpus/quadruples-validation.tsv": {"num_bytes": 163026560, "checksum": "802c26a7856b16f09e5380e54f115bee66d83539cc2f41bb39fbf651b99a31ed"}, "https://storage.googleapis.com/gresearch/kelm-corpus/quadruples-test.tsv": {"num_bytes": 163157370, "checksum": "d41be1cb6feed48d938136b8783ba67a9bfc262fc0614df6208937381de11e36"}}, "download_size": 1631259869, "post_processing_size": null, "dataset_size": 1678899973, "size_in_bytes": 3310159842}}
```
Comment (lhoestq) [inline datasets/kelm/dataset_infos.json:1]: can you regenerate this file ?
something is wrong in it (ex: `"builder_name": "ag_news"`)

**Review by lhoestq — COMMENTED** [review]
Looks very cool and clean :) 
thank you for adding this dataset

Can you just regenerate the dataset_infos.json file ?

File: datasets/kelm/dataset_infos.json (hunk 1 above)
Comment (joeddav) [reply datasets/kelm/dataset_infos.json]: Haha dang I'm 2 for 2 with that kind of error. At least I didn't merge this one 🤦‍♂️ fixed.

**Review by lhoestq — APPROVED** [review]
thanks ! Looks all good now

**Pull request merged**